In [69]:
# Importamos liberarías

import pandas as pd 
import numpy as np

In [70]:
df_ventas = pd.read_csv("ventas.csv", index_col=0)

df_ventas.head()

,ID_Producto,Fecha_Venta,Cantidad,Total
ID_Cliente,,,,
723,A1,2023-11-22,2,17.98
498,C3,2023-11-21,1,5.49
121,D4,2023-11-20,3,32.97
885,L12,2023-11-19,1,6.49
347,Q17,2023-11-18,2,7.98


In [71]:
df_ventas.shape

(100, 4)

In [72]:
df_ventas.info()

<class 'pandas.DataFrame'>
Index: 100 entries, 723 to 987
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   ID_Producto  100 non-null    str    
 1   Fecha_Venta  100 non-null    str    
 2   Cantidad     100 non-null    int64  
 3   Total        100 non-null    float64
dtypes: float64(1), int64(1), str(2)
memory usage: 3.9 KB


In [73]:
df_ventas.isna().sum()

ID_Producto    0
Fecha_Venta    0
Cantidad       0
Total          0
dtype: int64

In [74]:
df_ventas.duplicated().sum()

np.int64(0)

Posibles soluciones:

- on_bad_lines en read_csv
- copilot que nos ayude con el error
- hacerlo manual, abriendo el  fichero con visual, buscar patrón ", " y reemplazando por otro valor.

In [95]:
df_productos = pd.read_csv("productos.csv", index_col=0)

df_productos.head()

,Nombre_Producto,Categoría,Precio,Origen,Descripción
ItaliaID,,,,,
A1,Pizza Margherita,Platos Preparados,8.99,Italia,Clásica pizza italiana con tomate mozzarella f...
B2,Risotto de Champiñones,Platos Preparados,6.75,Italia,Risotto cremoso con champiñones frescos una de...
C3,Tiramisú,Postres,5.49,Italia,Postre clásico italiano con capas de bizcocho ...
D4,Panettone,Repostería,10.99,Italia,Pan dulce navideño italiano con frutas confita...
E5,Orecchiette,Productos Secos,4.29,Italia,Pequeñas pastas con forma de orecchiette ideal...


In [86]:
df_productos.shape

(33, 6)

In [97]:
df_productos["Origen"].unique()

<StringArray>
['Italia']
Length: 1, dtype: str

In [87]:
df_productos.info()

<class 'pandas.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ItaliaID         33 non-null     str    
 1   Nombre_Producto  33 non-null     str    
 2   Categoría        33 non-null     str    
 3   Precio           33 non-null     float64
 4   Origen           33 non-null     str    
 5   Descripción      33 non-null     str    
dtypes: float64(1), str(5)
memory usage: 1.7 KB


In [88]:
df_productos.isna().sum()

ItaliaID           0
Nombre_Producto    0
Categoría          0
Precio             0
Origen             0
Descripción        0
dtype: int64

In [89]:
df_productos.duplicated().sum()

np.int64(0)

In [90]:
df_clientes = pd.read_csv("clientes.csv", index_col=0)

df_clientes.head()

,first_name,last_name,email,gender,City,Country,Address
id,,,,,,,
1,Cheri,Dunsmore,cdunsmore0@instagram.com,Female,Palma De Mallorca,Spain,076 Rockefeller Crossing
2,Hunt,Bartomeu,hbartomeu1@nsw.gov.au,Male,Lugo,Spain,0046 Utah Junction
3,Michaeline,Paynton,mpaynton2@narod.ru,Female,NaN,Spain,0 Corry Crossing
4,Filmer,Eirwin,feirwin3@intel.com,NaN,Leon,Spain,5 American Ash Road
5,Tanhya,Lubbock,tlubbock4@huffingtonpost.com,Female,"Hospitalet De Llobregat, L'",Spain,9289 Merry Circle


In [100]:
df_clientes["Country"].value_counts(dropna=False)

Country
Spain    846
NaN      154
Name: count, dtype: int64

In [91]:
df_clientes.shape

(1000, 7)

In [92]:
df_clientes.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 1 to 1000
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   first_name  1000 non-null   str  
 1   last_name   1000 non-null   str  
 2   email       973 non-null    str  
 3   gender      923 non-null    str  
 4   City        876 non-null    str  
 5   Country     846 non-null    str  
 6   Address     959 non-null    str  
dtypes: str(7)
memory usage: 54.8 KB


In [93]:
df_clientes.isna().sum()

first_name      0
last_name       0
email          27
gender         77
City          124
Country       154
Address        41
dtype: int64

In [94]:
df_clientes.duplicated().sum()

np.int64(0)

# Para todos los ficheros:
    - Al usar el index_colum, las IDs pasan a ser los índices del dataframe.

# Para el fichero ventas:

    - La columna "Fecha_Venta", convertirla en dtype data.
    - Aparentemente, los valores son correctos, no hay nulos ni duplicados.

# Para el fichero productos:
    - Quitar tildes en los datos.
    - Aparentemente, los valores son correctos, no hay nulos ni duplicados.

# Para el fichero clientes:
    - Hay nulos, habría que gestionarlos:
        - email          27
        - gender         77
        - City          124
        - Country       154
        - Address        41
    - Aparentemente no hay duplicados

# <font color="red"> Para unificar los tres ficheros en uno, deberíamos de:</font>

- El fichero Ventas, tiene una ID de Producto y una ID de Cliente.
- El fichero Productos, tiene una ID de los distritos de Italia.
- El fichero Clientes, tiene una ID propia de clientes.

* Fichero Ventas --> con la Id de producto relaciona el fichero "productos"
* Fichero Ventas --> Con la ID de Cliente, se relaciona con el fichero "Clientes"


# Unificar los tres ficheros por filas usando las Ids